csharp/ql/test/resources/stubs/_frameworks/Microsoft.AspNetCore.App/Microsoft.AspNetCore.Authentication.Abstractions.cs

```diff
@@ -178,9 +162,8 @@ public interface IAuthenticationSchemeProvider
                 System.Threading.Tasks.Task<System.Collections.Generic.IEnumerable<Microsoft.AspNetCore.Authentication.AuthenticationScheme>> GetRequestHandlerSchemesAsync();
                 System.Threading.Tasks.Task<Microsoft.AspNetCore.Authentication.AuthenticationScheme> GetSchemeAsync(string name);
                 void RemoveScheme(string name);
-                bool TryAddScheme(Microsoft.AspNetCore.Authentication.AuthenticationScheme scheme) => throw null;
+                virtual bool TryAddScheme(Microsoft.AspNetCore.Authentication.AuthenticationScheme scheme) => throw null;
```

Comment:
Is this `virtual` correct?

Reply:
Appears so: https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.authentication.iauthenticationschemeprovider.tryaddscheme?view=aspnetcore-7.0#microsoft-aspnetcore-authentication-iauthenticationschemeprovider-tryaddscheme(microsoft-aspnetcore-authentication-authenticationscheme)